tds/src/main/java/thredds/server/viewer/Viewer.java

```diff
@@ -34,6 +33,13 @@
    * @param req the request
    * @return HTML fragment string
    */
-   String getViewerLinkHtml( Dataset ds, HttpServletRequest req);
+  String getViewerLinkHtml( Dataset ds, HttpServletRequest req);
+
+  /**
+   ** @param ds the dataset to view
+   * @param req the request
+   * @return HTML fragment string
+   */
+  ViewerLinkProvider.ViewerLink getViewerLink(Dataset ds, HttpServletRequest req);
```

Comment:
I realized after I started down this path that this is a public interface and will probably break compatibility for some people, but we do need a way to import the raw (not html-ed) link into the templates...

Reply:
I think we're ok. It should be pretty straight forward for people upgrading viewers from 4.6 to 5.0 to implement the new `getViewerLink` method.

Reply:
BTW - totally awesome that you are considering compatibility :-)